solidity/contracts/ECDSABackportRewards.sol

```diff
@@ -0,0 +1,127 @@
+  ▓▓▓▓▓▓▓▓▓▓▓▓▓▀        ▐▓▓▓▓▓▓▓▓▓▓▌        ▓▓▓▓▓▓▓▓▓▓▌        ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓
+  ▓▓▓▓▓▓▀▀▓▓▓▓▓▓▄       ▐▓▓▓▓▓▓▀▀▀▀         ▓▓▓▓▓▓▀▀▀▀         ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▀
+  ▓▓▓▓▓▓   ▀▓▓▓▓▓▓▄     ▐▓▓▓▓▓▓     ▓▓▓▓▓   ▓▓▓▓▓▓     ▓▓▓▓▓   ▐▓▓▓▓▓▌
+▓▓▓▓▓▓▓▓▓▓ █▓▓▓▓▓▓▓▓▓ ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓  ▓▓▓▓▓▓▓▓▓▓
+▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓ ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓  ▓▓▓▓▓▓▓▓▓▓
+
+                           Trust math, not hardware.
+*/
+
+pragma solidity 0.5.17;
+
+import "@keep-network/keep-core/contracts/Rewards.sol";
+import "./BondedECDSAKeepFactory.sol";
+import "./api/IBondedECDSAKeep.sol";
+
+contract ECDSABackportRewards is Rewards {
+    // BondedECDSAKeepFactory deployment date, May-13-2020 interval started.
+    // https://etherscan.io/address/0x18758f16988E61Cd4B61E6B930694BD9fB07C22F
+    uint256 internal constant bondedECDSAKeepFactoryDeployment = 1589408351;
+
+    // We are going to have one interval, with a weight of 100%.
+    uint256[] internal backportECDSAIntervalWeight = [100];
+
+    // Interval is the difference in time of creation between older and newer
+    // versions of BondedECDSAKeepFactory.
+    // Older: https://etherscan.io/address/0x18758f16988E61Cd4B61E6B930694BD9fB07C22F
+    // Newer: https://etherscan.io/address/0xA7d9E842EFB252389d613dA88EDa3731512e40bD
+    uint256 internal constant backportECDSATermLength = 123 days; // 10678946 sec
+
+    // There were 41 Keeps created by BondedECDSAKeepFactory : 0x18758f16988E61Cd4B61E6B930694BD9fB07C22F
+    // The last Keep was opened on May-17-2020
+    // https://etherscan.io/address/0x45A3cACA2F2a78A53607618651C86111c9720AA5
+    uint256 internal constant numberOfCreatedECDSAKeeps = 41;
+
+    // We allocate ecdsa backport rewards to all 41 keeps.
+    uint256
+        internal constant minimumECDSAKeepsPerInterval = numberOfCreatedECDSAKeeps;
+
+    BondedECDSAKeepFactory factory;
+
+    constructor(address _token, address payable _factoryAddress)
+        public
+        Rewards(
+            _token,
+            bondedECDSAKeepFactoryDeployment,
+            backportECDSAIntervalWeight,
+            backportECDSATermLength,
+            minimumECDSAKeepsPerInterval
+        )
+    {
+        factory = BondedECDSAKeepFactory(_factoryAddress);
+    }
+
+    function _getKeepCount() internal view returns (uint256) {
+        // Between May 13 2020 - Sep 14 2020 there were 41 keeps opened.
+        return numberOfCreatedECDSAKeeps;
+    }
+
+    function _getKeepAtIndex(uint256 i) internal view returns (bytes32) {
+        return bytes32(i);
+    }
+
+    function _getCreationTime(bytes32) internal view returns (uint256) {
+        // Assign each keep to the starting timestamp of its interval.
+        return startOf(0);
+    }
+
+    function _isClosed(bytes32) internal view returns (bool) {
+        // All keeps created between May 13 2020 - Sep 14 2020 are considered closed.
+        return true;
```

Comment:
This is a good decision but it took me a while to understand why you did it this way (we have 18 closed keeps and 23 active keeps). Maybe we can link to https://tbtc.network/news/2020-05-21-details-of-the-tbtc-deposit-pause-on-may-18-2020/ and explain closing some of those keeps is not easily achievable given the problem in the previous version of tBTC and do not want to block rewards for good stakers because of that bug.

Reply:
Added additional comment.